### PR TITLE
IAM pagination

### PR DESCRIPTION
**Pull request by jweibel22**
Add pagination to all calls to IAM to avoid errors when response is paged.
Changes log format to json so it can be properly parsed in Humio

### CODE REVIEW

File: internal/infrastructure/redshift/client.go [hunk 1]
```diff
@@ -142,6 +142,11 @@ func (c *Client) Schemas() ([]string, error) {
 }
 
 func (c *Client) ExternalSchemas() ([]redshift.ExternalSchema, error) {
+
+	if !c.externalSchemasSupported {
```
Comment (Crevil) [inline internal/infrastructure/redshift/client.go:146]: This doesn't look related?

**Review by Crevil — APPROVED** [review]
LGTM. Nice and clean 🚀

File: internal/infrastructure/redshift/client.go (hunk 1 above)
Comment (jweibel22) [reply internal/infrastructure/redshift/client.go:146]: no it wasn't. the integration tests are not run by CI :-( and got broken by the previous PR, so this is a fix for that.

Comment (Crevil) [reply internal/infrastructure/redshift/client.go:146]: 👍
